src/reversediff.jl

```diff
@@ -0,0 +1,14 @@
+using .ReverseDiff: ReverseDiff
+
+struct ReverseDiffBackend <: AbstractBackend end
+
+@primitive function pullback_function(ba::ReverseDiffBackend, f, xs...)
+    return (ws) -> begin
+        tape = ReverseDiff.InstructionTape()
+        x_tracked = ReverseDiff.track.(xs, Ref(tape))
+        y_tracked = f(x_tracked...)
+        ReverseDiff.deriv!.(y_tracked, ws...)
+        ReverseDiff.reverse_pass!(tape)
+        return ReverseDiff.deriv.(x_tracked)
+    end
```

Comment:
Is this the recommended form? In DiffEqSensitivity we use a slightly different set of instructions, see
https://github.com/SciML/DiffEqSensitivity.jl/blob/8601419ad910455ef8842597a8a13c615768477f/src/derivative_wrappers.jl#L406
(including `unseed!`, `input_hook`, `output_hook`)

Reply:
Perhaps, I wrote this a few months back and don't recall the reasons. Thanks the reference, I'll check.

Reply:
ReverseDiff is a little unclear about what constitutes its API. e.g. there's its [documented API](https://juliadiff.org/ReverseDiff.jl/api/), and the [`api` source directory](https://github.com/JuliaDiff/ReverseDiff.jl/tree/master/src/api). Some functions are declared in the latter but not documented (e.g. `input_hook` and `output_hook`), so they may be part of the API, while `unseed!`, for example, does not appear to be part of the API at all. Can you comment on why non-API functions were used in DiffEqSensitivity?

Reply:
I am not sure. Probably to support some more general cases than `seeded_reverse_pass!` does? CC @ChrisRackauckas

Reply:
It's to make it non-allocating and more generic. I got those hooks directly from Jarrett.

Reply:
Non-allocating would only be relevant for mutating `f!`, right? Currently I don't think mutating `f!` are supported by the AD API.